pulpcore/cli/rpm/common.py

```diff
@@ -1 +1 @@
-CHECKSUM_CHOICES = ("unknown", "md5", "sha1", "sha224", "sha256", "sha384", "sha512")
+CHECKSUM_CHOICES = ("sha256", "sha384", "sha512")
```

Comment:
Please find a way to support the old versions of pulp_rpm still.

Reply:
I'm about 99.9% certain that nobody has ever used this option in the first place.  Every RHEL since RHEL 6 has supported SHA256.  "unknown" actually shouldn't have been here in the first place.

But if I do - is there a pre-existing framework for tweaking the choices based on available versions of pulp/plugins?  I see `PluginRequirement` but it's not immediately clear if it would work in this context, as it has no "context" (object) available to operate off of.

https://github.com/pulp/pulp-cli/blob/589f9f4fdd0d7222141df6c53861fdfd990e4850/pulpcore/cli/rpm/repository.py#L139

Reply:
Let me see. First we should talk about pulp_rpm versions here. So let's assume these choices were removed in pulp_rpm version x.y.z.
Remove the `"unknown"` because that was never useful. But keep the reset, maybe add a note to the helpstring of the option. Then in the context object in the glue part there should be a `preprocess_entity` where based on the choice made, you can issue `self.pulp_ctx.needs_plugin(PluginRequirement("rpm", specifier=">=x.y.z", inverted=True, feature="deprecated checksums")`.